packages/prettier-plugin-java/src/printers/classes.js

```diff
@@ -88,63 +97,98 @@ class ClassesPrettierVisitor {
   }
 
   fieldDeclaration(ctx) {
-    return "fieldDeclaration";
+    const fieldModifiers = this.mapVisit(ctx.fieldModifier);
+    const unannType = this.visit(ctx.unannType);
+    const variableDeclaratorList = this.visit(ctx.variableDeclaratorList);
+
+    return join("", [
+      join(" ", fieldModifiers),
+      " ",
+      unannType,
+      " ",
+      variableDeclaratorList,
+      ";"
+    ]);
   }
 
   fieldModifier(ctx) {
-    return "fieldModifier";
+    if (ctx.annotation) {
+      return this.visit(ctx.annotation);
+    }
+    // public | protected | private | ...
+    return this.getSingle(ctx).image;
   }
 
   variableDeclaratorList(ctx) {
-    return "variableDeclaratorList";
+    const variableDeclarators = this.mapVisit(ctx.variableDeclarator);
+    const variableDeclaratorsSep = variableDeclarators.length > 0 ? ", " : "";
```

Comment:
why check length if according to the grammar there is always at least one?

Reply:
also why do we care about the separator? won't join on empty arraay return an empty Doc?